### PR TITLE
Fix min_seeding_time tag removal when max_seeding_time is -1

### LINKED ISSUE #597: [Bug]: MinSeedTimeNotReached tag not being applied correctly, causes torrent to pause and unpause every other run
### Describe the Bug

Recently added min_seeding_time to a share limit, it partially works in that the torrent hit the ratio and qbit paused it, then QM disabled the ratio and resumed it (but never added the MinSeedTimeNotReached tag), the next run though it sets back the ratio and a then qbit pauses it again....and repeat.

I tried adding the MinSeedTimeNotReached manually, and QM just removes it. The trace logs havent been helpful.

torrent with the issue is `mayor.of.kingstown.s03e05.1080p.web.h264-successfulcrab[EZTVx.to].mkv[eztvx.to]`

trace where it set the ratio and removed the tag i had manually added: https://privatebin.net/?5317f37d52b2c0a1#9mEhbDzfSF9gxCREG7u7VjTZiHAdaGZhWEyX6MU3BRNo
partial trace where it said it removed the ratio and said it added the tag but the tag wasnt added: https://privatebin.net/?5b69cc17542dc38d#BNWuQ1BEp5BYgEGfeob48hCiaoK5Vagq3kV14StTbaLa

also included the info level log of it removing the ratio one running and saying it was adding MinSeedTimeNotReached, then the next run saying it was both adding the MinSeedTimeNotReached and then setting the ratio back to 1

### Config

```yaml
# https://github.com/StuffAnThings/qbit_manage/wiki/Config-Setup
commands:
  dry_run: false
  cross_seed: false
  recheck: false
  cat_update: false
  tag_update: true
  rem_unregistered: false
  tag_tracker_error: true
  rem_orphaned: false
  tag_nohardlinks: true
  share_limits: true
  skip_qb_version_check: false
  skip_cleanup: false

qbt:
  host: qbittorrent:8080
  user:
  pass:

settings:
  force_auto_tmm: false
  tracker_error_tag: issue
  nohardlinks_tag: noHL
  share_limits_tag: ~share_limit
  share_limits_min_seeding_time_tag: MinSeedTimeNotReached
  share_limits_min_num_seeds_tag: MinSeedsNotMet
  share_limits_last_active_tag: LastActiveLimitNotReached
  cross_seed_tag: cross-seed
  cat_filter_completed: true
  share_limits_filter_completed: true
  tag_nohardlinks_filter_completed: true
  cat_update_all: true
  
directory:
  cross_seed:
  root_dir: /data/torrents/
  remote_dir: /data/torrents/
  recycle_bin: /data/torrents/.RecycleBin
  torrents_dir: /qbittorrent/data/BT_backup
  orphaned_dir: /data/torrents/orphaned_data

cat:
  movies: /data/torrents/movies
  tv: /data/torrents/tv
  audiobooks: /data/torrents/audiobooks
  books: /data/torrents/books
  manual: /data/torrents/manual
  music: /data/torrents/music
  prowlarr: /data/torrents/prowlarr
  cross-seed-link: /data/torrents/links

cat_change:

tracker:
  myanonamouse:
    tag: MaM
  torrentleech|tleechreload:
    tag: TorrentLeech
  other:
    tag: other

nohardlinks:
  # books: # Calibre doesn't hard link
  #   exclude_tags:
  #     - MaM
  #     - TorrentLeech
  audiobooks:
    # exclude_tags:
    #   - MaM
    #   - TorrentLeech
  movies:
    # exclude_tags:
    #   - MaM
    #   - TorrentLeech
  tv:
    # exclude_tags:
    #   - MaM
    #   - TorrentLeech
  manual:
    # exclude_tags:
    #   - MaM
    #   - TorrentLeech
  music:
    # exclude_tags:
    #   - MaM
    #   - TorrentLeech
  prowlarr:
    # exclude_tags:
    #   - MaM
    #   - TorrentLeech
  cross-seed-link:
    # exclude_tags:
    #   - MaM
    #   - TorrentLeech
  
share_limits:
  MaM:
    priority: 1
    include_all_tags:
      - MaM
    exclude_all_tags:
      - noHL
    max_ratio: -1
    max_seeding_time: -1
    # min_seeding_time: 75h
    cleanup: false
  MaM_noHL:
    priority: 2
    include_all_tags:
      - MaM
      - noHL
    max_ratio: -1
    max_seeding_time: -1
    # min_seeding_time: 75h
    cleanup: false
  TL:
    priority: 3
    include_all_tags:
      - TorrentLeech
    exclude_all_tags:
      - noHL
    max_ratio: -1
    max_seeding_time: -1
    # min_seeding_time: 11d
    cleanup: false
  TL_noHL:
    priority: 4
    include_all_tags:
      - TorrentLeech
      - noHL
    max_ratio: -1
    max_seeding_time: -1
    # min_seeding_time: 11d
    cleanup: false
  noHL:
    priority: 998
    include_all_tags:
      - noHL
      - other
    max_ratio: 1
    max_seeding_time: -1
    min_num_seeds: 2
    cleanup: true
  default:
    priority: 999
    max_ratio: 1
    max_seeding_time: -1
    min_seeding_time: 24h
    min_num_seeds: 5
    cleanup: true

recyclebin:
  enabled: true
  empty_after_x_days: 60
  save_torrents: true
  split_by_category: true

orphaned:
  empty_after_x_days: 60
  exclude_patterns:

apprise:
  api_url: http://apprise:8000
  notify_url: apprise://apprise:8000/default/?tags=qbitmanage

webhooks:
  error: apprise
  run_start: http://apprise:8000/notify/default?tag=qbitmanage_noise
  run_end: http://apprise:8000/notify/default?tag=qbitmanage_noise
  function:
    cross_seed: http://apprise:8000/notify/default?tag=qbitmanage_noise
    recheck: http://apprise:8000/notify/default?tag=qbitmanage_noise
    cat_update: http://apprise:8000/notify/default?tag=qbitmanage_noise
    tag_update: http://apprise:8000/notify/default?tag=qbitmanage_noise
    rem_unregistered: apprise
    tag_tracker_error: apprise
    rem_orphaned: http://apprise:8000/notify/default?tag=qbitmanage_noise
    tag_nohardlinks: http://apprise:8000/notify/default?tag=qbitmanage_noise
    share_limits: http://apprise:8000/notify/default?tag=qbitmanage_noise
    cleanup_dirs: apprise
```

### Logs

https://privatebin.net/?65ea07d7966ebdbd#GdNHtKfVkUnKsAmWgvKGyEuFHX8nRJpBFdcEMnFCpwpf

### Screenshots

_No response_

### Installation

Docker

### Version Number

4.1.7-develop1

### What branch are you on?

develop

### PR DESCRIPTION
**Pull request by ineednewpajamas**
## Description

Add logic for when max_seed_time is -1 to _has_reached_seeding_time_limit.

Fixes #597 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods
- [x] I have modified this PR to merge to the develop branch

### CODE REVIEW

**Comment by bobokun** [comment]
I've added an additional check similar for max_ratio. In your original test did you do a test to see if `min_seeding_time` was reached that it did delete the torrent correctly? Can you test with one torrent for this use case?

**Comment by ineednewpajamas** [comment]
Yep, here are the partial logs of when it got cleaned up while I was sleeping
```
|===================== Updating Share Limits for [Group default] [Priority 999] =====================|
|    Torrent Name: Mayor Of Kingstown S03E05 Iris 1080p AMZN WEB-DL DDP5 1 H 264-NTb[EZTVx.to].mkv[eztvx.to] |
|         Tracker: udp://tracker.opentrackr.org:1337                                                 |
|         Added Tag: ~share_limit_999.default                                                        |
|     Share Limit: Max Ratio = 1.0                                                                   |
|    Torrent Name: mayor.of.kingstown.s03e05.1080p.web.h264-successfulcrab[EZTVx.to].mkv[eztvx.to]   |
|         Tracker: udp://tracker.opentrackr.org:1337                                                 |
|         Added Tag: ~share_limit_999.default                                                        |
|     Share Limit: Max Ratio = 1.0                                                                   |
|======== Cleaning up torrents that have reached ratio/seed limit for default. Priority 999 =========|
|    Torrent Name: Mayor Of Kingstown S03E05 Iris 1080p AMZN WEB-DL DDP5 1 H 264-NTb[EZTVx.to].mkv[eztvx.to] |
|         Tracker: udp://tracker.opentrackr.org:1337                                                 |
|         Ratio vs Max Ratio: 1.34 >= 1.00                                                           |
|         Cleanup: True [Meets Share Limits]                                                         |
| Saving Torrent JSON file to /data/torrents/tv/.RecycleBin/torrents_json/Mayor Of Kingstown S03E05 Iris 1080p AMZN WEB-DL DDP5 1 H 264-NTb[EZTVx.to].mkv[eztvx.to].json |
| Backing up 8b83ade11f29fdc3ffb0b5c997aef715cbdc9273.torrent and .fastresume to /data/torrents/tv/.RecycleBin/torrents |
|         Deleted .torrent AND content files.                                                        |
|    Torrent Name: mayor.of.kingstown.s03e05.1080p.web.h264-successfulcrab[EZTVx.to].mkv[eztvx.to]   |
|         Tracker: udp://tracker.opentrackr.org:1337                                                 |
|         Ratio vs Max Ratio: 1.75 >= 1.00                                                           |
|         Cleanup: True [Meets Share Limits]                                                         |
| Saving Torrent JSON file to /data/torrents/tv/.RecycleBin/torrents_json/mayor.of.kingstown.s03e05.1080p.web.h264-successfulcrab[EZTVx.to].mkv[eztvx.to].json |
| Backing up 23574cf6c6095e70de486ece4d7b5373cfe73276.torrent and .fastresume to /data/torrents/tv/.RecycleBin/torrents |
|         Deleted .torrent AND content files.                                                        |
|====================================================================================================|
```

I added your change to my container and will have the next torrent to cleanup in about 15 hours if you wanted a second test.

File: modules/core/share_limits.py [hunk 1]
```diff
@@ -570,7 +570,7 @@ def _has_reached_seeding_time_limit():
         if last_active is not None:
             if not _has_reached_last_active_time_limit():
                 return body
-        if max_ratio is not None:
+        if max_ratio is not None or max_ratio != -1:
```
Comment (ineednewpajamas) [inline modules/core/share_limits.py:573]: Wouldn't this be an and statement?
For example -1 would eval as true for the "is not None", or am i still half asleep

Comment (bobokun) [reply modules/core/share_limits.py:573]: oops you're absolutely right. It's early for me so I'm the one that's half asleep 😅

**Comment by ineednewpajamas** [comment]
> Can you test with one torrent for this use case?

Are you referring to the use case for the code you added or just my original use case?

**Comment by bobokun** [comment]
> Are you referring to the use case for the code you added or just my original use case?

Just the original use case. I also realized the `max_ratio != -1` is not really necessary as well since it wouldn't ever satisfy any of the if conditions underneath. Either way it's probably better to add in to make things easier to read.

**Comment by ineednewpajamas** [comment]
Sounds good. Is what I provided good enough? otherwise the next test would be in around 14 hours or so.

**Review by bobokun — APPROVED** [review]

**Comment by bobokun** [comment]
Oops I linked to the wrong bug. It should be #597

**Pull request merged**